lib/rubocop/cop/minitest/assertion_in_hook.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # This cop checks for usage of assertions in hooks.
+      #
+      # @example
+      #   # bad
+      #   class FooTest < Minitest::Test
+      #     def setup
+      #       assert_equal(foo, bar)
+      #     end
+      #   end
+      #
+      #   # good
+      #   class FooTest < Minitest::Test
+      #     def test_something
+      #       assert_equal(foo, bar)
+      #     end
+      #   end
+      #
+      class AssertionInHook < Cop
```

Comment:
Can you rename to `AssertionInLifecycleHook`?

```suggestion
      class AssertionInLifecycleHook < Cop
```